### PR TITLE
Delay device opening

### PR DESCRIPTION
**Pull request by urso**
Reimplement the sniffer to delay:
- opening the sniffing device until sniffer.Run is executed.
- do some preliminary config validation on New only
- split file handling into separate sniffer handle type, to clean up the
  run loop a little

This change should help with packetbeat commands - other than run - complaining about not having permissions to open a device.

### CODE REVIEW

File: packetbeat/sniffer/file.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	data, ci, err := h.pcapHandle.ReadPacketData()
+	if err != nil {
+		if err != io.EOF {
+			return data, ci, err
+		}
+
+		h.pcapHandle.Close()
+		h.pcapHandle = nil
+
+		// TODO: Remove me.  still required to keep system tests working.
```
Comment (andrewkroh) [inline packetbeat/sniffer/file.go]: Would adding `-waitstop Xms` to the system test achieve a similar result?

Comment (urso) [inline packetbeat/sniffer/file.go]: waitstop is in seconds + I have had to add it to every test. The new publisher pipeline supports some wait stop behavior internally, but it's not used/exposed by any beat yet.

The `-waitstop` setting seems to be removed, in favor of a config file setting. I can try setting it to 300ms by default. But I didn't find this doesn't help much with test stability either. Ultimately we want #3404 (which should be possible with new publisher pipeline).

**Comment by urso** [comment]
I manually tested a few sub comments with and without root on OS X and linux. Also tested af_packet and pcap on linux still working.

This change guarantees `packetbeat test config`, `packetbeat devices`, `packetbeat export config` working without requiring rights on the device to be used by `run`. Drawback is, the `test config` does not check/validate the device actually exists.

**Comment by tsg** [comment]
jenkins, test it

**Pull request merged**